### PR TITLE
Create filters option on modal

### PR DESCRIPTION
**Pull request by corrigancd**
I've tested with: 
- filters created from WFS geoJsons 
- on no linked search dashboard
- on single search dashboard
- on Dashboard360
- in vis edit mode 

**Dashboard 360 - note that the geofilters of each search are treated independently:** 
CombineFiltersModalDB360

**Dashboard with no linked search - not that the filters applied of one search are applied to all:**
CombineFiltersModal

### CODE REVIEW

File: public/vislib/geoFilter.js [hunk 1]
```diff
@@ -177,30 +191,80 @@ define(function (require) {
...
-          onConfirm: () => {
-            _combineFilters(newFilter, existingFilter, field);
-          }
-        };
+      //counting total number of filters linked to the IndexPattern of NewFilter
+      const allFilters = [...queryFilter.getAppFilters(), ...queryFilter.getGlobalFilters()];
+      let numFiltersInIndexPattern = 0;
+      if (allFilters.length > 0) {
+        _.each(allFilters, filter => {
+          if (filter.meta.index === indexPatternName && isGeoFilter(filter, field)) {
```
Comment (szydan) [inline public/vislib/geoFilter.js]: I think indexPatternName is in reality indexPatternId
if true please rename everywhere

File: public/vislib/geoFilter.js [hunk 2]
```diff
@@ -177,30 +191,80 @@ define(function (require) {
...
+      };
 
-        confirmModal('How would you like this filter applied?', confirmModalOptions);
-      } else {
+      if (numFiltersInIndexPattern === 0 || numFiltersInIndexPattern >= 2) {
         _applyFilter(newFilter, field, indexPatternName);
-      }
-    }
 
+      } else if (_.isEqual(numFiltersInIndexPattern, 1)) {
```
Comment (szydan) [inline public/vislib/geoFilter.js]: numFiltersInIndexPattern === 1

File: public/vislib/geoFilter.js (hunk 1 above)
Comment (szydan) [inline public/vislib/geoFilter.js]: I think here it is NOT enough to check that the index pattern match 
There might be already 2 filters made from different maps which are based on the same index-pattern
I think we should also compare the meta_siren.vis (id + panelIndex)  to be sure that we set the existing filter to the correct one from these 2 (the one which was created from this vis before)

**Comment by corrigancd** [comment]
How it looks now. Also added a 3rd filter to a vis with success (not shown in GIF). 

CombineFiltersModalWithPanelIndexTakenIntoAccount

File: public/vislib/geoFilter.js [hunk 3]
```diff
@@ -181,33 +181,43 @@ define(function (require) {
...
       //counting total number of filters linked to the IndexPattern of NewFilter
       const allFilters = [...queryFilter.getAppFilters(), ...queryFilter.getGlobalFilters()];
+      const newFilterVisMeta = newFilter.meta._siren.vis;
       let numFiltersInIndexPattern = 0;
+
       if (allFilters.length > 0) {
         _.each(allFilters, filter => {
-          if (filter.meta.index === indexPatternName && isGeoFilter(filter, field)) {
-            existingFilter = filter;
+          const allFilters = filter.meta._siren.vis;
```
Comment (szydan) [inline public/vislib/geoFilter.js]: allFilters -> filterVisMeta

**Pull request merged**